test/test_torch.py

```diff
@@ -13052,8 +13067,23 @@ def check_sum_all(tensor):
         else:
             check_sum_all(torch.tensor([True, False, True], dtype=torch.bool, device=device))
 
-    def _test_memory_format_transformations(self, device, input_generator_fn, transformation_fn,
-                                            compare_data=True, default_is_preserve=False):
+    @onlyCPU
+    @dtypes(torch.bfloat16)
+    def test_sum_precision(self, device, dtype):
+        x1 = torch.rand(100, 100)
+        res1 = torch.sum(x1, 1)
+        x2 = x1.bfloat16()
+        res2 = torch.sum(x2, 1)
+        self.assertEqual(res1, res2, 2e-1)
+
+        x1 = torch.rand(100, 100, 100)
+        res1 = x1.sum(2).sum(1)
+        x2 = x1.bfloat16()
+        res2 = x2.sum(2).sum(1)
+        max_err = torch.abs((res1 - res2.float()) / res1).max()
```

Comment:
What exactly this test is trying to test? Why is there a truncation in the middle? Without truncation in the middle, the tolerance is too high (you should expect 1/256), truncation in the middle does not serve any discernible purpose.

Reply:
The code fragment is measuring the bfloat16 precision with more input data (100 -> 10000). If there is tow much input data, it is hard to give a fixed tolerance number(2^-7). For example, if all the sign(+/-) of input data is the same and all these data cannot be accurately represented as bfloat/half exactly, then the `sum` result between float and low precision(bfloat, half) can be very different and the difference should be extremely larger than 2^-7. So this test uses Relative-Tolerance number to measure precision.

Reply:
More input data is simulated by either creating a bigger tensor, or by reducing along multiple dimensions in one go. There is no reason to truncate in the middle other than to muddy the waters. Relative tolerance should be on the order of 1/256.